### ISSUE TITLE
Router gets in a funky state after deleting sample

### ISSUE THREAD

**Issue by pkalita-lbl** [microbiomedata/nmdc-field-notes#93]
Steps to reproduce:

1. From home page, navigate to a study with samples
2. From study page, navigate to an individual sample
3. From the sample page, use the ellipsis menu in the header to delete it. If successful, this should result in the user being navigated back to the study page.
4. From the study page, use the back button in the header to return to the home page

At this point, the router is a bit confused about which page should be displayed as evidenced by a ghostly version of the study page header displayed under the home page header:

![image](https://github.com/microbiomedata/nmdc-field-notes/assets/102547565/b31736e9-0059-4541-a27b-9366ad58e4e2)

Behind the scenes, Ionic keeps multiple pages in the DOM so that it can do the animated transitions between them. For example, at the end of step 2 above the relevant portion of the DOM look like this:

![image](https://github.com/microbiomedata/nmdc-field-notes/assets/102547565/b31a47b8-e697-4da8-af90-d779d5df843b)

You can see a stack of three `div` elements with the `ion-page` class. The bottom two are hidden, and the top two have the `can-go-back` class. I believe that means that tapping the "Back" button at that point would pop the top one off the stack and unhide the middle (or what will become the top) one.

This all seems to get jumbled up when we use the `router.push()` function provided by `useIonRouter`. This is the relevant portion of the DOM after the completion of step 4 above:
 
![image](https://github.com/microbiomedata/nmdc-field-notes/assets/102547565/0342f64e-6625-46e0-83dc-8460a9798afa)

We have two `div` elements with the `ion-page` class and **both** are visible.

I think the warrants a [deeper dive into Ionic routing](https://ionicframework.com/docs/react/navigation), but the solution might be to -- instead of always using `router.push()` after deleting an item -- use a combination of `router.canGoBack`, `router.goBack`, and `router.push` to **prefer** navigating backwards after the delete if possible and if not fallback to pushing a new route.

**Comment by pkalita-lbl** [comment]
After taking a second look at this, I think what's going on in Eric's use case is a little more understandable. 

My understanding is that each tab maintains its own separate history stack. The Guide and Settings tab "root" pages (`/in/guide` and `/in/settings`, respecitvely) both have a back button in the header with `defaultHref={"/"}`. The first time you visit `/in/guide` by switching tabs there will be no history in that tab's stack. Pressing the back button at that point will use the `defaultHref` and attempt to navigate to `/` with the "back"-style animation. I think things then get confused because `/` represents a route in a _different_ tab (i.e. a different history stack). So I think the solution for that issue is that the `/in/guide` and `/in/settings` should _not_ have back buttons at all in the header.

**Comment by eecavanna** [comment]
>My understanding is that each tab maintains its own separate history stack.

That is my understanding also. Right now, I am not seeing it explicitly stated in the Ionic docs, but I do see this statement on [the `ion-tab` page](https://ionicframework.com/docs/api/tab): "Each tab can contain a **top level navigation stack** for an app or a single view. An app can have many tabs, all with **their own independent navigation**" (it says _can_ as opposed to _will_).

---

Thanks for writing all of that out. I'm in favor of implementing the solution you proposed. Omitting the header's "Back" button when already at the base route of a tab's navigation tree makes sense to me.

**Comment by eecavanna** [comment]
> `@eecavanna`: Post information about what the DOM contains after step 3 in the previous post, like in the issue description.

Here's a screenshot showing what the DOM contains immediately after step 3.

![image](https://github.com/microbiomedata/nmdc-field-notes/assets/134325062/339ae6bc-941e-4a2c-9743-9e332b2301a9)

**Comment by eecavanna** [comment]
Here's a Git patch where I've implemented a solution like the one @pkalita-lbl described above (although I only implemented it for the "Guide" tab here).

```diff
diff --git a/src/Router.tsx b/src/Router.tsx
index 65586e9..15f2276 100644
--- a/src/Router.tsx
+++ b/src/Router.tsx
@@ -89,7 +89,7 @@ const Router: React.FC = () => {
 
             {/* GUIDE TAB ROUTES */}
             <AuthRoute exact path={paths.guide}>
-              <GuidePage />
+              <GuidePage basePath={paths.guide} />
             </AuthRoute>
 
             {/* SETTINGS TAB ROUTES */}
diff --git a/src/pages/GuidePage/GuidePage.tsx b/src/pages/GuidePage/GuidePage.tsx
index d19abad..210334e 100644
--- a/src/pages/GuidePage/GuidePage.tsx
+++ b/src/pages/GuidePage/GuidePage.tsx
@@ -9,15 +9,24 @@ import {
 } from "@ionic/react";
 import Checklist from "../../components/Checklist/Checklist";
 import ThemedToolbar from "../../components/ThemedToolbar/ThemedToolbar";
+import { useLocation } from "react-router-dom";
+
+interface Props {
+  basePath: string;
+}
+
+const GuidePage: React.FC<Props> = ({ basePath }) => {
+  const { pathname } = useLocation();
 
-const GuidePage: React.FC = () => {
   return (
     <IonPage>
       <IonHeader>
         <ThemedToolbar>
-          <IonButtons slot={"start"}>
-            <IonBackButton defaultHref={"/"}></IonBackButton>
-          </IonButtons>
+          {pathname === basePath ? null : (
+            <IonButtons slot={"start"}>
+              <IonBackButton defaultHref={"/"}></IonBackButton>
+            </IonButtons>
+          )}
           <IonTitle>Guide</IonTitle>
         </ThemedToolbar>
       </IonHeader>
```

With this patch in place, the "Guide" tab does not have a "Back" button in its header.

![image](https://github.com/microbiomedata/nmdc-field-notes/assets/134325062/fab31349-2ef4-44cd-8130-a06b4d747361)

**Comment by eecavanna** [comment]
Here's an even shorter patch 😅, which seems to me to produce the same outcome: 

```diff
diff --git a/src/pages/GuidePage/GuidePage.tsx b/src/pages/GuidePage/GuidePage.tsx
index d19abad..6392ad8 100644
--- a/src/pages/GuidePage/GuidePage.tsx
+++ b/src/pages/GuidePage/GuidePage.tsx
@@ -15,9 +15,9 @@ const GuidePage: React.FC = () => {
     <IonPage>
       <IonHeader>
         <ThemedToolbar>
-          <IonButtons slot={"start"}>
-            <IonBackButton defaultHref={"/"}></IonBackButton>
-          </IonButtons>
+            <IonButtons slot={"start"}>
+              <IonBackButton />
+            </IonButtons>
           <IonTitle>Guide</IonTitle>
         </ThemedToolbar>
       </IonHeader>
```

**Comment by pkalita-lbl** [comment]
I'm going to close this and say it has been fixed by the changes in #123.

I can no longer reproduce the issue using either of the methods outlined in this issue. Other ad-hoc testing didn't reveal any other ways to get into the described state.
